### PR TITLE
feat: add debug command

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
* **#10** <a href="https://app.graphite.dev/github/pr/skyleague/starchart/10?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9** <a href="https://app.graphite.dev/github/pr/skyleague/starchart/9?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7** <a href="https://app.graphite.dev/github/pr/skyleague/starchart/7?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pevisscher and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/commands/dev/function.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                    Effect: 'Allow',
+                    Action: [
+                        'iot:DescribeEndpoint',
+                        'iot:Connect',
+                        'iot:Subscribe',
+                        'iot:Publish',
+                        'iot:RetainPublish',
+                        'iot:Receive',
+                    ],
+                    Resource: '*',
```
Comment (MickVanDuijn) [inline src/commands/dev/function.ts:40]: do we want to restrict this more in the future? we could lock it down to a topic prefix if we want

File: src/commands/dev/handler.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+process.on('message', async ({ lambdaFn, event, context, rootDirectory }) => {
+    try {
+        const { tsImport } = await import('tsx/esm/api')
+        const loaded = await tsImport(`.${lambdaFn.tags.Path}.ts`, `${rootDirectory}/`)
+        if ('handler' in loaded && typeof loaded.handler === 'function' && context !== undefined) {
+            const response = await loaded.handler(event, { ...context, getRemainingTimeInMillis: () => 1000 })
+            process.send?.({ right: response })
+        } else {
+            process.send?.({ error: 'Handler function not found or context is undefined' })
```
Comment (MickVanDuijn) [inline src/commands/dev/handler.js]:
```suggestion
            process.send?.({ left: 'Handler function not found or context is undefined' })
```

File: src/commands/dev/handler.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        }
+    } catch (error) {
+        console.error(error)
+        process.send?.({ error: error.message })
```
Comment (MickVanDuijn) [inline src/commands/dev/handler.js]:
```suggestion
        process.send?.({ left: error.message })
```

File: src/commands/dev/function.ts (hunk 1 above)
Comment (pevisscher) [reply src/commands/dev/function.ts:40]: Yeah we can make it more restricted in the future. This helps it get going for now

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
### Merge activity

* **Aug 28, 3:35 PM GMT+2**: @pevisscher started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/skyleague/starchart/9).
* **Aug 28, 3:37 PM GMT+2**: [Graphite](https://app.graphite.dev/github/pr/skyleague/starchart/9) rebased this pull request as part of a merge.
* **Aug 28, 3:38 PM GMT+2**: @pevisscher merged this pull request with [Graphite](https://app.graphite.dev/github/pr/skyleague/starchart/9).

**Pull request merged**